Source/missiles.cpp

```diff
@@ -861,7 +858,9 @@ void PlayerMissile::hitMonster(int mid, int dam) const
 
 int TrapMissile::calculateCTH(Monster &monster) const
 {
-	return 90 - monster.mArmorClass - colMissile->_midist;
+	return (colMissile->_mitype == MIS_FIREWALL
+	        ? GenerateRnd(75) - monster.mLevel * 2

```

Comment:
This CTH vs monster is now calculated this way also for Firewalls originating from Oily shrine.
I am not sure why even Hellfire devs wanted special CTH vs monster only for Fire Ring trap in first place... Probably it was late found for this missile being TARGET_BOTH, which in all other cases is used for missiles originating from player.
Let me know if we want this special CTH for it at all. I think it could be removed as well.